### PR TITLE
Switch to spotless and format code

### PR DESCRIPTION
**Pull request by ZacSweers**
After merging, I can make a second PR setting up the format commit as an ignored revision to better protect history

Note that I haven't enabled licenses yet because there seems to be a bug in Spotless where it always overwrites the copyright year even if we had one set. https://github.com/diffplug/spotless/issues/677

### CODE REVIEW

**Comment by nedtwigg** [comment]
I might pause on merging this for a moment to evaluate `ratchetFrom 'origin/main'`, I'll explain in the issue you raised on Spotless.  Moshi is great btw, thanks!

File: adapters/src/main/java/com/squareup/moshi/adapters/Iso8601Utils.java [hunk 1]
```diff
@@ -26,11 +26,12 @@
...
  * https://github.com/FasterXML/jackson-databind/blob/67ebf7305f492285a8f9f4de31545f5f16fc7c3a/src/main/java/com/fasterxml/jackson/databind/util/ISO8601Utils.java
  *
- * Utilities methods for manipulating dates in iso8601 format. This is much much faster and GC
+ * <p>Utilities methods for manipulating dates in iso8601 format. This is much much faster and GC
  * friendly than using SimpleDateFormat so highly suitable if you (un)serialize lots of date
  * objects.
  *
- * Supported parse format: [yyyy-MM-dd|yyyyMMdd][T(hh:mm[:ss[.sss]]|hhmm[ss[.sss]])]?[Z|[+-]hh[:]mm]]
+ * <p>Supported parse format:
+ * [yyyy-MM-dd|yyyyMMdd][T(hh:mm[:ss[.sss]]|hhmm[ss[.sss]])]?[Z|[+-]hh[:]mm]]
```
Comment (swankjesse) [inline adapters/src/main/java/com/squareup/moshi/adapters/Iso8601Utils.java:34]: this one’s weird

File: adapters/src/test/java/com/squareup/moshi/adapters/PolymorphicJsonAdapterFactoryTest.java [hunk 2]
```diff
@@ -27,17 +30,17 @@
...
-    Moshi moshi = new Moshi.Builder()
-        .add(PolymorphicJsonAdapterFactory.of(Message.class, "type")
-            .withSubtype(Success.class, "success")
-            .withSubtype(Error.class, "error"))
-        .build();
+  @Test
+  public void fromJson() throws IOException {
+    Moshi moshi =
+        new Moshi.Builder()
+            .add(
```
Comment (swankjesse) [inline adapters/src/test/java/com/squareup/moshi/adapters/PolymorphicJsonAdapterFactoryTest.java:39]: ew

File: kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/api/AdapterGenerator.kt [hunk 3]
```diff
@@ -159,17 +170,17 @@ internal class AdapterGenerator(
...
-          prop.type.rawType() == JsonAdapter::class.asClassName()
-        }
-        .filter { prop -> prop.annotations.isNotEmpty() }
-        .mapTo(mutableSetOf()) { prop ->
-          QualifierAdapterProperty(
-              name = prop.name,
-              qualifiers = prop.annotations.mapTo(mutableSetOf()) { it.className }
-          )
-        }
+      .asSequence()
```
Comment (swankjesse) [inline kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/api/AdapterGenerator.kt:173]: still should be +4 on wrapped lines, right?

File: moshi/src/test/java/com/squareup/moshi/AdapterMethodsTest.java [hunk 4]
```diff
@@ -84,136 +86,157 @@
...
-      @FromJson @Uppercase String fromJson(@Uppercase String string) {
-        return string.substring(1, string.length() - 1);
-      }
-
-      @ToJson @Uppercase String toJson(@Uppercase String value) {
-        return "|" + value + "|";
-      }
-    }).add(new UppercaseAdapterFactory()).build();
+  @Test
+  public void toAndFromWithIntermediateWithQualifier() throws Exception {
```
Comment (swankjesse) [inline moshi/src/test/java/com/squareup/moshi/AdapterMethodsTest.java:169]: I think this is strictly worse

**Comment by swankjesse** [comment]
Also please don’t address my feedback above; I’m just ranting at bad choices made by the formatter

File: moshi/src/test/java/com/squareup/moshi/AdapterMethodsTest.java (hunk 4 above)
Comment (JakeWharton) [reply moshi/src/test/java/com/squareup/moshi/AdapterMethodsTest.java:169]: https://github.com/google/google-java-format/issues/342

File: adapters/src/test/java/com/squareup/moshi/adapters/PolymorphicJsonAdapterFactoryTest.java (hunk 2 above)
Comment (JakeWharton) [reply adapters/src/test/java/com/squareup/moshi/adapters/PolymorphicJsonAdapterFactoryTest.java:39]: you get used to it

https://github.com/google/google-java-format/wiki/The-Rectangle-Rule

File: adapters/src/main/java/com/squareup/moshi/adapters/Iso8601Utils.java (hunk 1 above)
Comment (JakeWharton) [reply adapters/src/main/java/com/squareup/moshi/adapters/Iso8601Utils.java:34]:
```
jshell> " * Supported parse format: [yyyy-MM-dd|yyyyMMdd][T(hh:mm[:ss[.sss]]|hhmm[ss[.sss]])]?[Z|[+-]hh[:]mm]]".length()
$1 ==> 101
```

**Comment by JakeWharton** [comment]
:shipit:. Switched over Retrofit a while ago and I don't care. Same for using ktlint on a bunch of projects. Whevs.

**Comment by swankjesse** [comment]
SGTM. Thanks Zach!

File: kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/api/AdapterGenerator.kt (hunk 3 above)
Comment (ZacSweers) [reply kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/api/AdapterGenerator.kt:173]: Ktlint doesn't support continuation indent anymore, and the compose and kotlin people sold me recently on it being a "you get used to it" thing. The kotlin plugin is iffy on its support of it too

Comment (JakeWharton) [reply kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/api/AdapterGenerator.kt:173]: Yep, you won't notice in a week.

**Pull request merged**